### PR TITLE
The Test fetches the latest glibc repo and runs the tests in it.

### PR DESCRIPTION
**Pull request by santhosh93**
If the execution fails with non-zero status then output is parsed for the
failures and displayed to the user.

Signed-off-by: Santhosh G santhog4@linux.vnet.ibm.com

### CODE REVIEW

**Comment by santhosh93** [comment]
@ldoktor : Test does not hang after applying the changes from PR https://github.com/avocado-framework/avocado/pull/1521

File: toolchain/glibc.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from avocado.core import data_dir
+
+
+class glibc(Test):
+    def setUp(self):
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+        deps = ['gcc', 'make', 'gawk']
+        self.tmpdir = data_dir.get_tmp_dir()
+        self.build_dir = self.params.get('build_dir', default=self.tmpdir)
```
Comment (PraveenPenguin) [inline toolchain/glibc.py]: can this be `self.build_dir = self.params.get('build_dir', default=data_dir.get_tmp_dir())`

**Comment by santhosh93** [comment]
@PraveenPenguin : Have incorporated the changes mentioned in the PR https://github.com/avocado-framework-tests/avocado-misc-tests/pull/39
Please, take a look at it and merge it.

File: toolchain/glibc.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        url = 'https://github.com/bminor/glibc/archive/master.zip'
+        tarball = self.fetch_asset("glibc.zip", locations=[url], expire='0d')
```
Comment (PraveenPenguin) [inline toolchain/glibc.py]: please provide some valid days for expire

**Comment by PraveenPenguin** [comment]
@santhosh93 some minor comment please have a look , waiting for V2 
Thanks !!

File: toolchain/glibc.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        archive.extract(tarball, self.srcdir)
+        self.srcdir = os.path.join(self.srcdir, "glibc-master")
+        os.chdir(self.build_dir)
+        process.run(self.srcdir + '/configure --prefix=%s' % self.build_dir,
```
Comment (PraveenPenguin) [inline toolchain/glibc.py]: if you do not want to retain `os.chdir(self.build_dir) ` please use `os.path.join() to add dir and file`

File: toolchain/glibc.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        deps = ['gcc', 'make', 'gawk']
+        self.tmpdir = data_dir.get_tmp_dir()
+        self.build_dir = self.params.get('build_dir', default=self.tmpdir)
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        url = 'https://github.com/bminor/glibc/archive/master.zip'
+        tarball = self.fetch_asset("glibc.zip", locations=[url], expire='0d')
+        archive.extract(tarball, self.srcdir)
+        self.srcdir = os.path.join(self.srcdir, "glibc-master")
```
Comment (santhosh93) [inline toolchain/glibc.py]: @narasimhan-v : Normally, we need to be in build directory and use the files in source directory to configure. So changed to build directory path.

File: toolchain/glibc.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        os.chdir(self.build_dir)
```
Comment (santhosh93) [inline toolchain/glibc.py]: @narasimhan-v : Normally, we need to be in build directory and use the files in source directory to configure. So changed to build directory path.

Comment (santhosh93) [inline toolchain/glibc.py]: @narasimhan-v : Precisely, If we are not in build_dir , make will fail.

**Comment by PraveenPenguin** [comment]
@santhosh93  please check https://github.com/avocado-framework/avocado/pull/1521 , as ldoktor  code not going to part of avocado 

Regards
Praveen

File: toolchain/glibc.py (hunk 5 above)
Comment (narasimhan-v) [inline toolchain/glibc.py]: @santhosh93 For clarification, you are specifying the `--prefix` parameter in configure, and still this is needed ?

Comment (santhosh93) [inline toolchain/glibc.py]: @narasimhan-v : yes without it make will build it in default directory and not in build directory

**Comment by ldoktor** [comment]
@PraveenPenguin in the end it's probably going to be merged, but as optional argument. Let's wait how the https://github.com/avocado-framework/avocado/pull/1538 turns out.

**Comment by PraveenPenguin** [comment]
@ldoktor sure thanks

File: toolchain/glibc.py (hunk 5 above)
Comment (narasimhan-v) [inline toolchain/glibc.py]: @santhosh93 ok

**Comment by PraveenPenguin** [comment]
@santhosh93 ping

**Comment by santhosh93** [comment]
Opened a new PR in https://github.com/avocado-framework-tests/avocado-misc-tests/pull/158
So closing it

**Pull request closed without merging**